### PR TITLE
Let Ed25519SimpleAddressKeyringEntry.createIdentity take an index 

### LINKED ISSUE #371: Let Ed25519SimpleAddressKeyringEntry.createIdentity take an index argument
A simple address takes an index `i` as defined in https://github.com/iov-one/iov-core/blob/master/docs/KeyBase.md#simple-addresses

This is in intermediate step for #302

### PR DESCRIPTION
**Pull request by webmaster128**
Resolves #371

### CODE REVIEW

File: packages/iov-bns/src/client.spec.ts [hunk 1]
```diff
@@ -52,12 +52,12 @@ describe("Integration tests with bov+tendermint", () => {
   // max open connections??? (but 900 by default)
   const tendermintUrl = "ws://localhost:22345";
 
-  const userProfile = async (): Promise<UserProfile> => {
+  async function userProfileWithFaucet(): Promise<UserProfile> {
```
Comment (ethanfrey) [inline packages/iov-bns/src/client.spec.ts]: you really don't like my `async () =>` abuse do you?

Comment (ethanfrey) [reply packages/iov-bns/src/client.spec.ts]: better name in any case

File: packages/iov-core/src/writer.spec.ts [hunk 2]
```diff
@@ -46,15 +46,6 @@ describe("IovWriter", () => {
       return profile;
     };
 
-    // will make identities if needed.
```
Comment (ethanfrey) [inline packages/iov-core/src/writer.spec.ts]: yeah, you just got rid of this nasty logic. 🎉

File: packages/iov-keycontrol/src/keyring-entries/slip10.ts [hunk 3]
```diff
@@ -170,7 +170,7 @@ export class Slip10KeyringEntry implements KeyringEntry {
     this.labelProducer.update(label);
   }
 
-  public async createIdentity(): Promise<LocalIdentity> {
+  public async createIdentity(_: number): Promise<LocalIdentity> {
```
Comment (ethanfrey) [inline packages/iov-keycontrol/src/keyring-entries/slip10.ts]: hmm... didn't you make this take optional keypair in other pr?

File: packages/iov-bns/src/client.spec.ts (hunk 1 above)
Comment (webmaster128) [reply packages/iov-bns/src/client.spec.ts]: I am trying hard to get used to it but it is still taking me considerably longer to parse that code when reading

Comment (ethanfrey) [reply packages/iov-bns/src/client.spec.ts]: The only time I think it really shines are either inline functions or partial ones...

```
const parse = (config: Config) => (item: Item) => {
  // do somethng with item based on config values
}
const parsed = items.map(parse(myConfig));
```

type def on top line reads almost like Haskell ;)

**Pull request merged**